Mathlib/Geometry/Manifold/Algebra/Monoid.lean

```diff
@@ -514,8 +514,10 @@ section
 variable {𝕜 : Type*} [NontriviallyNormedField 𝕜] {E : Type*} [NormedAddCommGroup E]
   [NormedSpace 𝕜 E]
 
-instance hasSmoothAddSelf : SmoothAdd 𝓘(𝕜, E) E :=
-  ⟨by rw [← modelWithCornersSelf_prod]; exact contDiff_add.contMDiff⟩
+instance hasSmoothAddSelf : SmoothAdd 𝓘(𝕜, E) E := by
+  constructor
+  rw [← modelWithCornersSelf_prod, chartedSpaceSelf_prod]
+  exact contDiff_add.contMDiff
```

Comment:
Is this change needed? Reading just the source, I would think that `constructor` creates two sub-goals, each line solves one, and you could condense them as the original proof.

Reply:
:bulb: Right... and you have a point. Then my only request is to use focus the two sub-goals.

Reply:
There are never two separate goals (see original proof)

Reply:
`SmoothAdd` has only one field that is not solved by type-class inference.

Reply:
So my mental model of constructor was wrong. (Reading the documentation, that indeed makes sense!) Thanks for pointing this out; sorry for the noise.